### PR TITLE
PR for step 4: Data base implementation

### PR DESCRIPTION
**Pull request by iggyray**
## What

- Migrated database to sqlite
- Created `items` and `category` tables (schema in items.db)
- Refactored `GET` & `POST` functions to be compatible with sqlite while retaining functionality
- Implemented `/search` endpoint that returns database records that match keyword
- Updated `.gitignore` so that `mercari.sqlite3` is not commited

## CHECKS :warning:

Please make sure you are aware of the following.

- [x] **The changes in this PR doesn't have private information

### CODE REVIEW

File: python/main.py [hunk 1]
```diff
@@ -48,23 +91,27 @@ async def add_item(name: str = Form(...), category: str = Form(...), image: Uplo
...
+        return "database is empty"
+
+@app.get("/search")
+def get_searched_response(keyword: str):
+    matches = searchForDbItem(keyword)
+    if len(matches) > 0:
+        returnItems = formatItemsForReturn(matches)
+        return returnItems
+    else:
+        return f"No matches found for: {keyword}"
```
Comment (mutokrm) [inline python/main.py]: Consider returning status code 404 if you want to see no matching records as an error!
Ref: 
https://fastapi.tiangolo.com/advanced/additional-status-codes/

File: python/main.py [hunk 2]
```diff
@@ -48,23 +91,27 @@ async def add_item(name: str = Form(...), category: str = Form(...), image: Uplo
...
 @app.get("/items")
 def get_items_reponse():
-    allItems = getItems()
-    return allItems
+    allItems = getDbItems()
+    if len(allItems) > 0:
+        returnItems = formatItemsForReturn(allItems)
+        return returnItems
+    else:
+        return "database is empty"
```
Comment (mutokrm) [inline python/main.py]: Same here

File: python/main.py [hunk 3]
```diff
@@ -48,23 +91,27 @@ async def add_item(name: str = Form(...), category: str = Form(...), image: Uplo
...
-    allItems["items"].append(newItem)
-
-    with open(fileName, 'w') as itemsFile:
-        json.dump(allItems, itemsFile)
+    postDbItem(name, category, hashedImageName)
 
     return {"message": f"item received with name: {name}, category: {category}, image: {hashedImageName}"}
 
 @app.get("/items")
 def get_items_reponse():
```
Comment (mutokrm) [inline python/main.py]: nit: you might wanna change the function names. FastAPI has a thing called SwaggerUI where you can visually check and test your endpoints. The names of the functions serve as a description of the endpoints 👍

Ref:
https://fastapi.tiangolo.com/tutorial/first-steps/

**Review by mutokrm — APPROVED** [review]
Added some small comments! I'll give you LGTM since there's no significant issues 👍

File: python/main.py (hunk 3 above)
Comment (iggyray) [reply python/main.py]: @walnut07 thank you for the comments! Changes made and merged!

**Pull request merged**